writeletter.rs

```diff
@@ -40,28 +40,38 @@ fn main() {
     std::io::stdin().read_line(&mut age).unwrap();
     let age: i32 = age.trim().parse().unwrap();
 
-    if age <= 0 || age >= 110 {
-        println!("you're kidding!");
-    } else {
-        println!(
-            "I heard you just had a birthday and you are {} years old.",
-            age
-        );
+    match age {
+        -100..=-1 | 110..=200 => {
```

Comment:
This has two problems.

1. According to the specification, age 0 and age 110 should fall into this ("you're kidding") category.

    > If age is 0 or less or 110 or more, call simple_error("you're kidding!")

1. With these ranges, -200 would be a valid age. Use [`i32::MAX`](https://doc.rust-lang.org/std/primitive.i32.html#associatedconstant.MAX) and [`i32::MIN`](https://doc.rust-lang.org/std/primitive.i32.html#associatedconstant.MIN) so that only ages between 1 and 109 are considered valid.